shared/utility/math/filter/UKF.hpp

```diff
@@ -177,27 +187,33 @@ namespace utility::math::filter {
             covariance_weights.fill(1.0 / (2.0 * covariance_sigma_weight));
             covariance_weights[0] = lambda / covariance_sigma_weight + (1.0 - (alpha * alpha) + beta);
 
-            set_state(initial_mean, initial_covariance);
+            return set_state(initial_mean, initial_covariance);
         }
 
-        void set_state(StateVec initial_mean, StateMat initial_covariance) {
+        [[nodiscard]] bool set_state(StateVec initial_mean, StateMat initial_covariance) {
             mean       = initial_mean;
             covariance = initial_covariance;
 
             // Calculate our sigma points
-            sigma_mean   = mean;
-            sigma_points = generate_sigma_points(mean, covariance, covariance_sigma_weight);
+            sigma_mean                  = mean;
+            auto&& points_chole_success = generate_sigma_points(mean, covariance, covariance_sigma_weight);
+
+            sigma_points = points_chole_success.first;
 
             // Reset our state for more measurements
             covariance_update = covariance_weights.asDiagonal();
             d.setZero();
             centred_sigma_points = sigma_points.colwise() - sigma_mean;
+            return points_chole_success.second;
```

Comment:
I think we need an early termination/reset here as well. If `generate_sigma_points` failed then `centred_sigma_points is a useless calculation?
```suggestion
            auto&& sigma_points = generate_sigma_points(mean, covariance, covariance_sigma_weight);

            // Reset our state for more measurements
            covariance_update = covariance_weights.asDiagonal();
            d.setZero();
            centred_sigma_points = sigma_points.first.colwise() - sigma_mean;
            return sigma_points.second;
```

Reply:
Is this suggestion saying that the class variable `sigma_points` should be changed to a `std::pair`?

That being said, this `std::pair` thing isn't clean at all. I am wondering if there's a better way to do it

Reply:
Oh, didnt realise that was a class variable. I just hate that variable name lol

Reply:
@CMurtagh-LGTM has suggested using `std::optional<sigma_points_type>` or throwing an exception if this is expected to happen more often. How often are we expecting to have cholesky decomposition errors - as in, is it common enough to warrant throwing errors?

tagging @TrentHouliston too. come join the c++ party :sweat_smile:

Reply:
Everywhere that `generate_sigma_points` is called, the first thing we do is assign the result to `sigma_points`. You can always just return a `bool` from `generate_sigma_points` and `generate_sigma_points` can set `sigma_points` internally if the decomposition succeeded

Reply:
Roger. I'll have a go with that.

Reply:
Ahh, it already throws lol. My code was poorly thought through hahahha